src/vivarium/interface/cli.py

```diff
@@ -129,21 +128,25 @@ def run(
     results_root.mkdir(parents=True, exist_ok=False)
 
     configure_logging_to_file(output_directory=results_root)
-    shutil.copy(model_specification, results_root / "model_specification.yaml")
 
     output_data = {"results_directory": str(results_root)}
     input_data = {}
     if artifact_path:
         input_data["artifact_path"] = artifact_path
     override_configuration = {"output_data": output_data, "input_data": input_data}
 
-    main = handle_exceptions(run_simulation, logger, with_debugger)
-    finished_sim = main(model_specification, configuration=override_configuration)
+    sim = SimulationContext(
+        model_specification=model_specification, configuration=override_configuration
+    )
+    with open(f"{results_directory}/model_specification.yaml", "w") as f:
+        yaml.dump(sim.model_specification.to_dict(), f)
+
+    handle_exceptions(sim.run_simulation, logger, with_debugger)()
```

Comment:
I think I like this better as two lines like it was before. It's easy to miss the trailing `()` when reading.